### PR TITLE
fix(doc): Disable SQL strict mode

### PR DESCRIPTION
**Pull request by lpinsivy**
## Description

Explain how to disable SQL strict mode

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [x] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [x] 19.10.x (master)

### CODE REVIEW

File: doc/en/installation/common/sql_strict_mode.rst [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    Centreon started the compatibility with SQL strict mode but not all components are ready yet. It is mandatory to
+    disable the strict mode if you use MariaDB >= 10.2.4 or MySQL >= 5.7.5. In this case, connect to your DBMS with
+    root account and execute the following SQL request: ::
+        
+        # SET sql_mode = 'NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION';
+        # SET GLOBAL sql_mode = 'NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION';
+    
+    or modify the */etc/my.cnf.d/centreon.cnf* file to add in the '[server]' section the following line: ::
+        
+        sql_mode = 'NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION'
```
Comment (sc979) [inline doc/en/installation/common/sql_strict_mode.rst]: does not work on deported mySql 8 DB

File: doc/en/release_notes/centreon-19.10/centreon-19.10.0-beta.1.rst [hunk 2]
```diff
@@ -33,7 +33,7 @@ Technical
 ---------
 
 * Compatibility with rrdtool >= 1.7.x (PR/#7676)
-* Compatibility with MariaDB/MySQL STRICT mode (PR/#7544)
+* Start compatibility with MariaDB/MySQL STRICT mode - in progres (PR/#7544)
```
Comment (sc979) [inline doc/en/release_notes/centreon-19.10/centreon-19.10.0-beta.1.rst]:
```suggestion
* Start compatibility with MariaDB/MySQL STRICT mode - in progress (PR/#7544)
```

File: doc/fr/installation/common/sql_strict_mode.rst [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    Centreon a démarré sa compatibilité avec le mode strict SQL. Cependant, tous ses composants ne sont pas encore
+    prêts. C'est pourquoi il est impératif de désactiver le mode strict SQL si vous utilisez MariaDB >= 10.2.4 ou MySQL
+    >= 5.7.5. Dans ce cas, exécutez les commandes suivantes : ::
+        
+        # SET sql_mode = 'NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION';
+        # SET GLOBAL sql_mode = 'NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION';
+    
+    ou modifiez le fichier */etc/my.cnf.d/centreon.cnf* pour ajouter à la section '[server]' la ligne suivante : ::
+        
+        sql_mode = 'NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION'
```
Comment (sc979) [inline doc/fr/installation/common/sql_strict_mode.rst]: ne fonctionne pas sur MySQL 8

File: doc/fr/release_notes/centreon-19.10/centreon-19.10.0-beta.1.rst [hunk 4]
```diff
@@ -33,7 +33,7 @@ Technical
 ---------
 
 * Compatibility with rrdtool >= 1.7.x (PR/#7676)
-* Compatibility with MariaDB/MySQL STRICT mode (PR/#7544)
+* Start compatibility with MariaDB/MySQL STRICT mode - in progres (PR/#7544)
```
Comment (sc979) [inline doc/fr/release_notes/centreon-19.10/centreon-19.10.0-beta.1.rst]:
```suggestion
* Start compatibility with MariaDB/MySQL STRICT mode - in progress (PR/#7544)
```

File: doc/en/release_notes/centreon-19.10/centreon-19.10.0-beta.1.rst (hunk 2 above)
Comment (kduret) [inline doc/en/release_notes/centreon-19.10/centreon-19.10.0-beta.1.rst]:
```suggestion
* Start compatibility with MariaDB/MySQL STRICT mode - in progress (PR/#7544)
```

File: doc/fr/release_notes/centreon-19.10/centreon-19.10.0-beta.1.rst (hunk 4 above)
Comment (kduret) [inline doc/fr/release_notes/centreon-19.10/centreon-19.10.0-beta.1.rst]:
```suggestion
* Start compatibility with MariaDB/MySQL STRICT mode - in progress (PR/#7544)
```

File: doc/en/installation/common/install_packages.rst [hunk 5]
```diff
@@ -86,11 +88,6 @@ Run the commands::
 .. note::
     **centreon-database** package installs a database server optimized for use with Centreon.
 
-.. note::
```
Comment (sc979) [inline doc/en/installation/common/install_packages.rst]: we talked about strict mode in beta mode, didn't we ?

File: doc/en/installation/common/sql_strict_mode.rst (hunk 1 above)
Comment (kduret) [reply doc/en/installation/common/sql_strict_mode.rst]: are you sure ? cause it looks the same here : https://kevdees.com/disable-mysql-strict-mode-and-no_zero_date-errors-in-homestead/

Comment (jlamotte) [reply doc/en/installation/common/sql_strict_mode.rst]: 2019-10-10T11:38:56.347451Z 0 [ERROR] [MY-000077] [Server] /usr/sbin/mysqld: Error while setting value 'NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION' to 'sql_mode'.

Comment (jlamotte) [reply doc/en/installation/common/sql_strict_mode.rst]: [root@mysql mysql]# mysql -V
mysql  Ver 8.0.17 for Linux on x86_64 (MySQL Community Server - GPL)

Comment (kduret) [reply doc/en/installation/common/sql_strict_mode.rst]: indeed, "NO_AUTO_CREATE_USER" does not exist anymore since MySQL 8.0.11

Comment (sc979) [reply doc/en/installation/common/sql_strict_mode.rst]: furthermore, we really need to split the documentation for mariaDB and MySQL, has this kind of behavior will be more frequent. And incoherences are already multiples in the current page (some command line are MySQL8 specific, other mariaDB)
@jlamotte has already began to check it

Comment (jlamotte) [reply doc/en/installation/common/sql_strict_mode.rst]: In mysql 8 for disable strict mode SET GLOBAL sql_mode = 'NO_ENGINE_SUBSTITUTION'

Comment (lpinsivy) [reply doc/en/installation/common/sql_strict_mode.rst]: you right

File: doc/en/installation/common/install_packages.rst (hunk 5 above)
Comment (lpinsivy) [reply doc/en/installation/common/install_packages.rst]: in Release note

**Comment by sc979** [comment]
Indeed if the chosen password for the MySQL8's 'centreon' user is low, centreon won't be able to create the new user and no error will be displayed.
As currently, we don't check for strong password in the form.
Resulting in a failed step without notice.
The script, then, will try to continue its work but will remain stuck in an infinite loop waiting for something without results.

**Comment by sc979** [comment]
"As currently, we don't check for strong password in the form." -> the install script, needs to check if the user is created and the privileges are working fine, before moving to the next step

File: doc/en/installation/common/sql_strict_mode.rst [hunk 6]
```diff
@@ -0,0 +1,29 @@
+.. warning::
+    Centreon started the compatibility with SQL strict mode but not all components are ready yet. It is mandatory to
+    disable the strict mode if you use MariaDB >= 10.2.4 or MySQL >= 5.7.5.
+
```
Comment (sc979) [inline doc/en/installation/common/sql_strict_mode.rst]: "Notice :You can enable the strict mode for testing purpose, but you shouldn't do so on your production environment as it's still in a beta stage"

File: doc/fr/installation/common/install_packages.rst [hunk 7]
```diff
@@ -103,6 +99,13 @@ Puis créer un utilisateur **root** distant : ::
     Remplacez **IP** par l'adresse IP publique du serveur Centreon et **PASSWORD**
     par le mot de passe de l'utilisateur **root**.
 
+.. warning::
+    MySQL >= 8 requière un mot de passe fort. Utilisez des lettres minuscules et majuscules ainsi que des caractères
```
Comment (sc979) [inline doc/fr/installation/common/install_packages.rst]:
```suggestion
    MySQL >= 8 requiert un mot de passe fort. Utilisez des lettres minuscules et majuscules ainsi que des caractères
```

File: doc/fr/installation/common/install_packages.rst [hunk 8]
```diff
@@ -130,13 +135,22 @@ La base de données MySQL doit être disponible pour pouvoir continuer l'install
 
 Pour les systèmes CentOS / RHEL en version 7, il est nécessaire de modifier
 la limitation **LimitNOFILE**. Changer cette option dans /etc/my.cnf NE
```
Comment (sc979) [inline doc/fr/installation/common/install_packages.rst]:
```suggestion
la limitation **LimitNOFILE**. Changer cette option dans /etc/my.cnf
```

File: doc/fr/installation/common/install_packages.rst [hunk 9]
```diff
@@ -130,13 +135,22 @@ La base de données MySQL doit être disponible pour pouvoir continuer l'install
...
-fonctionnera PAS: ::
+fonctionnera PAS.
```
Comment (sc979) [inline doc/fr/installation/common/install_packages.rst]:
```suggestion
ne fonctionnera PAS.
```

File: doc/fr/installation/common/install_packages.rst [hunk 10]
```diff
@@ -130,13 +135,22 @@ La base de données MySQL doit être disponible pour pouvoir continuer l'install
...
     # echo -ne "[Service]\nLimitNOFILE=32000\n" | tee /etc/systemd/system/mariadb.service.d/limits.conf
     # systemctl daemon-reload
     # systemctl restart mysql
 
+**Pour MySQL** : ::
+
+    # mkdir -p  /etc/systemd/system/mysqld.service.d
+    # echo -ne "[Service]\nLimitNOFILE=32000\n" | tee /etc/systemd/system/mysqld.service.d/limits.conf
+    # systemctl daemon-reload
+    # systemctl restart mysql
```
Comment (sc979) [inline doc/fr/installation/common/install_packages.rst]:
```suggestion
    # systemctl restart mysqld
```

File: doc/en/installation/common/install_packages.rst [hunk 11]
```diff
@@ -125,14 +130,21 @@ We recommend using MariaDB for your database because it is open source. Ensure
...
+    # echo -ne "[Service]\nLimitNOFILE=32000\n" | tee /etc/systemd/system/mariadb.service.d/limits.conf
+    # systemctl daemon-reload
+    # systemctl restart mysql
+
+**For MySQL**: ::
+
+    # mkdir -p  /etc/systemd/system/mysqld.service.d
+    # echo -ne "[Service]\nLimitNOFILE=32000\n" | tee /etc/systemd/system/mysqld.service.d/limits.conf
+    # systemctl daemon-reload
+    # systemctl restart mysql
```
Comment (sc979) [inline doc/en/installation/common/install_packages.rst]:
```suggestion
    # systemctl restart mysqld
```

File: doc/fr/installation/common/install_packages.rst [hunk 12]
```diff
@@ -196,6 +212,6 @@ doivent être exécutées : ::
     # systemctl start rh-php72-php-fpm
     # systemctl start httpd24-httpd
     # systemctl start mysqld
-    # systemctl start cbd
+    # systemctl start centeron
```
Comment (sc979) [inline doc/fr/installation/common/install_packages.rst]:
```suggestion
    # systemctl start centreon
```

File: doc/fr/installation/common/sql_strict_mode.rst [hunk 13]
```diff
@@ -0,0 +1,30 @@
+.. warning::
+    Centreon a démarré sa compatibilité avec le mode strict SQL. Cependant, tous ses composants ne sont pas encore
+    prêts. C'est pourquoi il est impératif de désactiver le mode strict SQL si vous utilisez MariaDB >= 10.2.4 ou MySQL
+    >= 5.7.5.
+
```
Comment (sc979) [inline doc/fr/installation/common/sql_strict_mode.rst]: ajout d'un "notice - mode beta" voir EN

**Comment by lpinsivy** [comment]
> Reverse "Setting the PHP time zone" and "Configuring/disabling the firewall" step
> Otherwise, php will not start and it will throw error.

Of course PHP will start with firewalld ebabled

**Pull request merged**